src/servers/trigger.ts

```diff
@@ -107,24 +107,14 @@ export async function onchainTransactionEventHandler(tx) {
   } else {
     // incoming transaction
 
-    let user
-    try {
-      user = await User.findOne(
-        { "onchain.address": { $in: tx.output_addresses } },
-        { lastIPs: 0, lastConnection: 0 },
-      )
-      if (!user) {
-        //FIXME: Log the onchain address, need to first find which of the tx.output_addresses belongs to us
-        onchainLogger.fatal(`No user associated with the onchain address`)
-        return
-      }
-    } catch (error) {
-      onchainLogger.error(
-        { error },
-        "issue in onchainTransactionEventHandler to get User id attached to output_addresses",
-      )
-      throw error
-    }
+    const walletsRepo = WalletsRepository()
+
+    // TODO: tx.output_addresses pass an array of address
+    // in the unlikely event multiple destination addresses are belong to the Galoy wallet
+    // only the first one will be notified
+    const wallet_ = await walletsRepo.findByAddress(tx.output_addresses)
+    if (wallet_ instanceof Error) return
+    const wallet = wallet_
```

Comment:
after instanceof validation wallet can be used as wallet type.. you dont have to re-assign it